### ISSUE TITLE
Drop support for godep, glide, dep, vendoring

### ISSUE THREAD

**Issue by jameshochadel** [cloudfoundry/go-buildpack#374]
Go modules, Go's official solution to dependency management, have been [on by default since Go 1.16](https://go.dev/blog/go116-module-changes). Per [the docs](https://docs.cloudfoundry.org/buildpacks/go/index.html) and the output I see when running `cf push`, the buildpack still contains several dependency management solutions that predate modules:

* [Godep](https://github.com/tools/godep), archived on Github since 2019
* [Glide](https://github.com/Masterminds/glide), not updated since 2019
* [dep](https://github.com/golang/dep), deprecated and archived since 2020

Additionally, the buildpack checks for a `vendor` folder, but go modules does not vendor dependencies by default; instead, it downloads dependencies go `$GOPATH/pkg/mod`. Vendoring is still possible, but in practice, I see few projects do it.

I propose removing the old dependency management solutions in favor of Go modules, and propose detecting the Go buildpack by checking for the presence of a `go.mod` file.

**Comment by jameshochadel** [comment]
👋 Just curious — Would the maintainers accept a PR to add this feature?

**Comment by robdimsdale** [comment]
I think this is reasonable, and I'd love to drop dead code paths. It would also reduce the surface area we have to test, which would be a benefit to the maintainers, too.

But, I know that CF buildpacks are used by a huge variety of consumers with varying degrees of progression, and I worry that if we drop support for an old vendoring model somewhere down the line this will break our consumers.

**Comment by jameshochadel** [comment]
Fair enough. Do you know if CF collects any telemetry we could use to help decide whether or not to drop the old methods?
